internal/services/workloads/workloads_sap_three_tier_virtual_instance_resource.go

```diff
@@ -0,0 +1,1961 @@
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringIsNotEmpty,
+											},
+										},
+									},
+								},
+
+								"database_type": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									ForceNew:     true,
+									ValidateFunc: validation.StringInSlice(sapvirtualinstances.PossibleValuesForSAPDatabaseType(), false),
+								},
+
+								"disk_volume_configuration": {
+									Type:     pluginsdk.TypeSet,
+									Optional: true,
+									ForceNew: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"volume_name": {
+												Type:     pluginsdk.TypeString,
+												Required: true,
+												ForceNew: true,
+												ValidateFunc: validation.StringInSlice([]string{
+													"backup",
+													"hana/data",
+													"hana/log",
+													"hana/shared",
+													"os",
+													"usr/sap",
+												}, false),
+											},
+
+											"number_of_disks": {
+												Type:         pluginsdk.TypeInt,
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.IntAtLeast(1),
+											},
+
+											"size_in_gb": {
+												Type:     pluginsdk.TypeInt,
+												Required: true,
+												ForceNew: true,
+											},
+
+											"sku_name": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringInSlice(sapvirtualinstances.PossibleValuesForDiskSkuName(), false),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+
+					"resource_names": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"application_server": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									ForceNew: true,
+									MaxItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"availability_set_name": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringLenBetween(1, 80),
+											},
+
+											"virtual_machine": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"data_disk_names": {
+															Type:     pluginsdk.TypeMap,
+															Optional: true,
+															ForceNew: true,
+															Elem: &pluginsdk.Schema{
+																Type:         pluginsdk.TypeString,
+																ValidateFunc: validation.StringLenBetween(1, 80),
+															},
+														},
```

Comment:
Currently this will be specified as below:  
```
        data_disk_names = {
            hanaData   = "hanadatapr0,hanadatapr1"
            hanaLog    = "hanalogpr0,hanalogpr1,hanalogpr2"
            usrSap     = "usrsappr0"
            hanaShared = "hanasharedpr0,hanasharedpr1"
          }
```
However, specifying comma separated values is complicated for users and it won't be easy to interpolate this. Instead we could represent this differently in the schema to make it more native to how terraform normally handles this. So the user could specify this field like this instead:

```
data_disk_name {
disk_type = hanaData
names = ["hanadatapr0", "hanadatapr1"]
}

data_disk_name {
disk_type = hanaLog
names = ["hanalogpr0", "hanalogpr1", "hanalogpr2"]
}
```

Reply:
Updated